bitwarden_license/src/Sso/Controllers/AccountController.cs

```diff
@@ -148,37 +146,65 @@ IActionResult invalidJson(string errorMessageKey, Exception ex = null)
                     }
-                    return invalidJson(errorKey, translatedException.ResourceNotFound ? ex : null);
+                    return InvalidJson(errorKey, translatedException.ResourceNotFound ? ex : null);
                 }
+
+                var ssoToken = new SsoToken()
+                {
+                    DomainHint = domainHint,
+                    OrganizationId = organization.Id
+                };
+
+                var tokenable = new SsoTokenable(ssoToken);
+                var token = _dataProtector.Protect(tokenable);
+
+                var json = Json(new
+                {
+                    token
+                });
+
+                return json;
             }
             catch (Exception ex)
             {
-                return invalidJson("PreValidationError", ex);
+                return InvalidJson("PreValidationError", ex);
             }
-
-            // Everything is good!
-            return new EmptyResult();
         }
 
         [HttpGet]
         public async Task<IActionResult> Login(string returnUrl)
         {
             var context = await _interaction.GetAuthorizationContextAsync(returnUrl);
-            if (context.Parameters.AllKeys.Contains("domain_hint") &&
-                !string.IsNullOrWhiteSpace(context.Parameters["domain_hint"]))
+
+            if (!context.Parameters.AllKeys.Contains("domain_hint") &&
+                string.IsNullOrWhiteSpace(context.Parameters["domain_hint"]))
             {
-                return RedirectToAction(nameof(ExternalChallenge), new
-                {
-                    scheme = context.Parameters["domain_hint"],
-                    returnUrl,
-                    state = context.Parameters["state"],
-                    userIdentifier = context.Parameters["session_state"]
-                });
+                throw new Exception(_i18nService.T("NoDomainHintProvided"));
             }
-            else
+                
+            var ssoToken = context.Parameters[SsoToken.TokenName];
+            var domainHint = context.Parameters["domain_hint"];
+            var organization = await _organizationRepository.GetByIdentifierAsync(domainHint);
+
+            if (organization == null)
             {
-                throw new Exception(_i18nService.T("NoDomainHintProvided"));
+                return InvalidJson("OrganizationNotFoundByIdentifierError");
+            }
+
+            var tokenable = _dataProtector.Unprotect(ssoToken);
+
+            if (!tokenable.TokenIsValid(new SsoToken() { OrganizationId = organization.Id, DomainHint = domainHint}))
+            {
+                return Unauthorized();
             }
+
+            return RedirectToAction(nameof(ExternalChallenge), new
+            {
+                scheme = context.Parameters["organizationId"],
+                returnUrl,
+                state = context.Parameters["state"],
+                userIdentifier = context.Parameters["session_state"]
```

Comment:
Code Style: since we're moving this block, even though it was not correctly formatted before, can we please add a trailing comma per our code-style guidelines here as well?
```suggestion
                userIdentifier = context.Parameters["session_state"],
```